src/lib.rs

```diff
@@ -363,23 +354,42 @@ where
     /// storage.
     ///
     /// Returns the number of bytes pushed to `buf`
-    pub fn retreive_frames(&mut self, storage: &mut S, buf: &mut [u8]) -> Result<usize, Error> {
-        let read_len = self.helper.read_slice(storage, buf)?;
+    pub fn retrieve_frames(&mut self, storage: &mut S, buf: &mut [u8]) -> Result<usize, Error> {
```

Comment:
Is there a reason for the `retrieve_frames_helper` vs `retrieve_frames` ? Couldn't they just be combined into `retrieve_frames`?

Reply:
It was easier for [defmt-print](https://github.com/eupn/defmt-print/blob/4fcaf42a41b14041c357bf22fb1109753ac963a0/src/main.rs#L70) to reuse the code from this crate if we extract frames reading logic into a static helper function like this. This way it won't require `defmt-print` to instantiate `LogManager` instance (as well as `bbqueue` buffer) only for a purpose of reading log frames.

Reply:
Ahh, okay :+1: Fair enough.

As long as there was reasoning behind it :)